### PR TITLE
pios_bmp280: Make taskless.

### PR DESCRIPTION
**Pull request by glowtape**
Saves some memory I guess. Works on my Seppuku. Don't have a target with the baro on I2C.

### CODE REVIEW

File: flight/PiOS/Common/pios_bmp280.c [hunk 1]
```diff
@@ -540,37 +526,38 @@ int32_t PIOS_BMP280_Test()
...
+	read_adc_result = PIOS_BMP280_ReadADC();
+	PIOS_BMP280_ReleaseDevice();
 
-		float calc_alt = 44330.0f * (1.0f - powf(data.pressure / BMP280_P0, (1.0f / 5.255f)));
+	if (read_adc_result) {
+		return false;
+	}
 
-		data.temperature = ((float) dev->compensatedTemperature) / 256.0f / 100.0f;
+	struct pios_sensor_baro_data *data = (struct pios_sensor_baro_data *)output;
```
Comment (tracernz) [inline flight/PiOS/Common/pios_bmp280.c:552]: What's the purpose of this var? Doesn't seem to add anything.

Comment (glowtape) [reply flight/PiOS/Common/pios_bmp280.c:552]: Right, should probably put it directly in the function header.

Comment (tracernz) [reply flight/PiOS/Common/pios_bmp280.c:552]: Ahhh, it's for the cast. No, it's fine as is. 👍

**Review by tracernz — APPROVED** [review]

**Comment by tracernz** [comment]
Tested this with #2157 on Seppuku. It was happy with healthy sensors. Don't have a really good way to check sensor error paths since we killed the Disco targets.

**Pull request merged**